### ISSUE TITLE
Make --join the default

### ISSUE THREAD

**Issue by stephentoub** [dotnet/BenchmarkDotNet#1574]
When running lots of benchmarks, the first time I run them I invariably forget to include `--join` on my command line invocation, resulting in the benchmark results being spread across a long history of spewed output.  I then take the time to run all the benchmarks again, this time with `--join`, as doing so is much less time consuming then finding all the results and manually cobbling them together into one table.  Personally, I've never once wanted the non-joined behavior, and it'd be great from my perspective if it were the default.

**Comment by Scooletz** [comment]
I agree with what @stephentoub stated above. I don't rerun my benchmarks and I do spend my time on

> finding all the results and manually cobbling them together into one table

but having `--join` as a default would be great.

**Comment by adamsitnik** [comment]
@AndreyAkinshin any thoughts on that?

if we are going to implement it we should remember to allow the users to disable this behavior by passing `--join false` which is going to require a workaround (https://github.com/tmds/Tmds.LinuxAsync/pull/29)

**Comment by YegorStepanov** [comment]
Bump, this is the top 6 problem by likes.

We can work around the bug by making `.Join` is `Nullable<bool>` (like Adam did in another repository).

```C#
"" -> true
"--join" -> true
"--join=true" -> true
"--join=false" -> false
```

**Comment by timcassell** [comment]
Join as default would be nice, but I think it should work properly for all cases first. Are you still going to work on #2163 @YegorStepanov?

**Comment by YegorStepanov** [comment]
> It should work properly for all cases first

In what cases does it not work?

> Are you still going to work on https://github.com/dotnet/BenchmarkDotNet/pull/2163

Yep, sorry for not answering for a long time!

**Comment by timcassell** [comment]
> In what cases does it not work?

The issues that #2163 claims to fix (multiple types).

**Comment by AndreyAkinshin** [comment]
I agree with @timcassell: it makes sense to merge #2163 first.
